### PR TITLE
Add player health, enemy collision, and update enemy sprite

### PR DESCRIPTION
**Pull request by mjarvis1997**
- added health bar
- player takes damage on collision with enemy
- new enemy sprite

### CODE REVIEW

File: Signals.gd [hunk 1]
```diff
@@ -0,0 +1,3 @@
+extends Node
+
+signal player_health_changed(value)
```
Comment (mjarvis1997) [inline Signals.gd:3]: I'm loving this global singleton stuff, it makes it very simple to share events between nodes without coupling their logic

File: enemies/enemy_1/enemy_1.gd [hunk 2]
```diff
@@ -0,0 +1,16 @@
+extends Area2D
+
+const SPEED = 250.0
+const COLLISION_DAMAGE = 20
+
+func _physics_process(delta):
+	var player_location = Globals.player.global_position
+	var movement_vector = delta * SPEED
+	global_position = global_position.move_toward(player_location, movement_vector)
```
Comment (mjarvis1997) [inline enemies/enemy_1/enemy_1.gd:9]: same thing you were doing before, just using `move_toward`, I changed this from PlayerCharacter2D to Area2D (generic collision object). i'm not sure what the advantages of the character node are, maybe we switch back at some point

File: enemies/enemy_1/enemy_1.gd [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+
+func _on_body_entered(body):
+	print('enemy body entered!')
+	if body.is_in_group("Player"):
+		body.take_damage(COLLISION_DAMAGE)
+	pass # Replace with function body.
```
Comment (mjarvis1997) [inline enemies/enemy_1/enemy_1.gd:16]: collision logic

File: player/health/health.gd [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	health = max_health
+	pass # Replace with function body.
+	
+func take_damage(damage):
+	health -= damage
+	print("Taking damage, current health: ", health)
+	Signals.emit_signal("player_health_changed", health)
```
Comment (mjarvis1997) [inline player/health/health.gd:16]: emitting to the global signal... its so slick (emit the signal)

File: player/player.gd [hunk 5]
```diff
@@ -11,30 +11,19 @@ func _ready():
...
 func _on_fire_button():
 	var new_projectile = Mini_Rocket.instantiate()
 	add_child(new_projectile)
 	
 func _input(event):
-	if event is InputEventJoypadButton:
-		if event.is_action_pressed("player_shoot"):
-			shoot.emit(Mini_Rocket, rotation, $ProjectileStart.global_position);
+	if event.is_action_pressed("player_shoot"):
+		shoot.emit(Mini_Rocket, rotation, $ProjectileStart.global_position);
```
Comment (mjarvis1997) [inline player/player.gd:24]: shooting working for me now on controller and keyboard

File: ui/health_bar.gd [hunk 6]
```diff
@@ -0,0 +1,12 @@
+extends Node2D
+
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	# Connect to global signal so we can respond when its emitted
+	Signals.player_health_changed.connect(self.update_health_bar)
```
Comment (mjarvis1997) [inline ui/health_bar.gd:7]: connecting local function to global signal (respond to the signal)

**Review by CSpiegelhalter — APPROVED** [review]

**Pull request merged**